src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -84,7 +81,35 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_Parameter1>$(NeutralLanguage)</_Parameter1>
       </AssemblyAttribute>
     </ItemGroup>
+  </Target>
+
+  <!-- 
+    To allow version changes to be respected on incremental builds (e.g. through CLI parameters),
+    create a hash of all assembly attributes so that the lock file name will change with the calculated
+    assembly attribute values and msbuild will then execute CoreGenerateAssembly to generate a new file.
+  -->
+  <Target Name="_CreateGeneratedAssemblyInfoInputsCacheFile"
+          DependsOnTargets="_CalculateAssemblyAttributes"
+          BeforeTargets="CoreGenerateAssemblyInfo">
```

Comment:
I'm asking because a huge
```xml
<Target Name="GenerateAssemblyInfo" … DependsOnTargets="PrepareForBuild;GetAssemblyVersion;GetAssemblyAttributes;_CreateGeneratedAssemblyInfoInputsCacheFile;CoreGenerateAssemblyInfo" … />
```
works as well and I wanted to make sure it doesn't do so by accident. But it somehow feels morally wrong.